### PR TITLE
hsmtool.c - Added new method to enable creation of hsm_file from cmd-line args

### PR DESCRIPTION
**Pull request by bstin**
Creating a hsm_file with `hsmtool` is currently an interactive process in terminal where user is expected to make choices for language, typing in seed phrase, etc. This works well, but is difficult to utilize for automated installs (ie. Docker, unattended-installs, etc). 

This adds a new method: `generatehsmbyargs` that allows the user to input all required args on command-line and generate the hsm_file without terminal interaction. 

Also the documentation was `lightning-hsmtool.8.md` updated to include new method. 

All existing functionality was preserved, this only added a new method.

### CODE REVIEW

**Comment by rustyrussell** [comment]
Sorry about the delay reviewing!

I prefer to extend generatehsm to take the cmdline args if they are present?  That ensures that the language list stays consistent, for example.

**Comment by niftynei** [comment]
@bstin any update on this?

**Comment by vincenzopalazzo** [comment]
>I prefer to extend generatehsm to take the cmdline args if they are present? That ensures that the language list stays consistent, for example.


For compatibility with another mobile wallet where people can import this list of words, I would like to have just English as an option, using different languages can cause lack with other wallets

**Comment by rustyrussell** [comment]
I rewrote this to extend the generatehsm command line instead.

File: tests/test_wallet.py [hunk 1]
```diff
@@ -1383,6 +1383,27 @@ def test_hsmtool_generatehsm(node_factory):
...
     assert l1.info['id'] == '02244b73339edd004bc6dfbb953a87984c88e9e7c02ca14ef6ec593ca6be622ba7'
+    l1.stop()
+
+    # We can do the entire thing non-interactive!
+    os.remove(hsm_path)
+    subprocess.check_output(["tools/hsmtool",
+                             "generatehsm", hsm_path,
+                             "en",
+                             "abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon about"])
+    assert open(hsm_path, "rb").read().hex() == "5eb00bbddcf069084889a8ab9155568165f5c453ccb85e70811aaed6f6da5fc1"
```
Comment (vincenzopalazzo) [inline tests/test_wallet.py:1394]: I was more happy to have a separate test, but anyway 😄

**Review by vincenzopalazzo — APPROVED** [review]
ACK 40e2f1e83f46bd721e6465ac4180dafff1b91977

**Review by niftynei — COMMENTED** [review]
ACK

File: tools/hsmtool.c [hunk 2]
```diff
@@ -43,7 +43,7 @@ static void show_usage(const char *progname)
 	       "<path/to/hsm_secret>\n");
 	printf("	- guesstoremote <P2WPKH address> <node id> <tries> "
 	       "<path/to/hsm_secret>\n");
-	printf("	- generatehsm <path/to/new/hsm_secret>\n");
+	printf("	- generatehsm <path/to/new/hsm_secret> [<language_id> <word list> [<password>]");
```
[the remark this answers is not in the document]
Comment (niftynei) [reply tools/hsmtool.c]: edit: pushed up fix for this!

**Comment by rustyrussell** [comment]
Rebased...

**Pull request merged**